code/modules/mob/living/basic/bots/bot_hud.dm

```diff
@@ -46,13 +46,18 @@
 
 	if(isnull(ai_controller))
 		return
-
+	// This doesn't do what you think it does, it removes the path images, but does not remove it from huds
 	clear_path_hud()
```

Comment:
It's not a merge blocker, but instead of the comment could this proc be renamed to something that better describes its function? There's only 6 hits for it in the code, and 1 of them is the proc definition itself.

My hope is that it would make this proc more self-documenting so this is less likely to happen again.

Reply:
I don't actually know what it does, which is why i didn't comment it.

Reply:
in theory the removal of hud client images could be moved into it, which would make ti actually do what it says on the tin.

Reply:
okay lookinga t it more it basically just makes the client path animate out of existence, which will mean every hud that sees it will see the diagnostic path fade out.

Reply:
I've changed it so clear_path_Hud actually clears the hud client images as well

This makes more sense to me on thinking about it because it's called in a few places including login (when a user takes over) (reset when a bot path is reset) and destroy.

So this will prevent all leaks